Civi/Api4/Generic/Action/Basic/Get.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Civi\Api4\Generic\Action\Basic;
+
+use Civi\Api4\Generic\Action\AbstractGet;
+use Civi\Api4\Generic\Result;
+
+/**
+ * Retrieve items based on criteria specified in the 'where' param.
+ *
+ * Use the 'select' param to determine which fields are returned, defaults to *.
+ */
+class Get extends AbstractGet {
+  use \Civi\Api4\Generic\Action\Traits\ArrayQueryTrait;
+
+  /**
+   * @var callable
+   */
+  private $getter;
+
+  public function __construct($entity, $getter = NULL) {
+    parent::__construct($entity);
+    if ($getter) {
+      $this->getter = $getter;
+    }
+    else {
+      $this->getter = [$this, 'getObjects'];
+    }
+  }
+
+  /**
+   * We expect an array of arrays from the getter.
+   * We pass it the params array in case it wants to do any pre-filtering for performance.
+   *
+   * @param \Civi\Api4\Generic\Result $result
+   */
+  public function _run(Result $result) {
+    $values = call_user_func($this->getter, $this->getParams());
```

Comment:
So... when I was first learning about class reflection... it was drilled into me that reflective operations (like `getParams()`) are more expensive. The trade-off is great and worthy for admin/developer tasks (ex: *browsing API Explorer*) but should be avoided for runtime tasks (ex: *importing a batch of 1000 records*). I know Moore's Law doesn't look kindly on such things... but it's an old habit and a simple change...

Anyway... I'd suggest just passing `$this` instead of `$this->getParams()`. Observations:

* It avoids a round of reflection/serialization.
* Most getters probably won't do anything with the params anyway.
* If a getter does need to do something with params, then it's nice for the dev to have type-hinting (`/** @param BasicGet $apiCall */`).

Reply:
@totten I've just updated with this and your other suggestions.